api/v1/controllers/samples.js

```diff
@@ -79,6 +82,46 @@ function doFindSample(req, res, next, resultObj, cacheKey, cacheExpiry) {
   });
 } // doFindSample
 
+/**
+ * Avoid non-running Collectors to send in Samples.
+ *
+ * Rejects if it's a Generator request (header.IsGenerator) and its current
+ * Collector has status != RUNNING - response back as 403 Forbidden.
+ *
+ * Resolves when it's a valid Generator or request is not from a Generator.
+ *
+ * @param req
+ * @returns {Promise}
+ */
+function validateNonRunningCollectors(req) {
+  return new Promise((resolve, reject) => {
+    if (!req.headers.IsGenerator) return resolve();
+
+    const param = { key: { value: req.headers.TokenName } };
+    u.findByKey(generators, param)
+      .then((generator) => {
+        if (!generator.currentCollector) {
+          const noCurrentCollector = 'Cannot accept Generator ' +
+            generator.name + ' samples without current collector';
+          return reject(
+            new apiErrors.ForbiddenError(
+              { explanation: noCurrentCollector, }
+            ));
+        } else if (generator.currentCollector &&
+          generator.currentCollector.status !== 'Running') {
+          const notRunning = 'Cannot accept samples from Collector ' +
+            generator.currentCollector.name + ' with status "' +
+            generator.currentCollector.status + '"';
+          return reject(new apiErrors.ForbiddenError(
+            { explanation: notRunning, })
+          );
+        }
+
+        return resolve();
+      }).catch((err) => reject(err));
```

Comment:
@iamigo by definition all findBy* methods return ResourceNotFoundError, just added a catch. 

Did not find an easy way to change the Token name to force the failure automatically, however, did the test manually changing the value in runtime when debugging.